### PR TITLE
exp/lighthorizon: add horizon web docker/k8s deployment

### LINKED ISSUE #4493: Dev deployment of Horizon Light
### What problem does your feature solve?
Access to a common/shared instance of Horizon Light Web server pointed at pubnet indexes and hosted on cloud infra. 
Provides a demoable instance.

### What would you like to see?
A public url that can be accessed from anywhere that has the HorizonLight web service behind it.

On Later iteration, add in CI pipeline to automate updating the HorizonLight instance when git branch changes.

### What alternatives are there?
do demos from local workstations running HorizonLight web server.

### PR DESCRIPTION
**Pull request by sreuland**
<details>
  <summary>PR Checklist</summary>
  
### PR Structure

* [x] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [ ] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [ ] This PR's title starts with name of package that is most changed in the PR, ex.
  `services/friendbot`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
* [ ] I've updated any docs ([developer docs](https://developers.stellar.org/api/), `.md`
  files, etc... affected by this change). Take a look in the `docs` folder for a given service,
  like [this one](https://github.com/stellar/go/tree/master/services/horizon/internal/docs).

### Release planning

* [ ] I've updated the relevant CHANGELOG ([here](services/horizon/CHANGELOG.md) for Horizon) if
  needed with deprecations, added features, breaking changes, and DB schema changes.
* [ ] I've decided if this PR requires a new major/minor version according to
  [semver](https://semver.org/), or if it's mainly a patch change. The PR is targeted at the next
  release branch if it's not a patch change.
</details>

### What

Added docker and k8s deployment wrapper for the lighthorizon web process. 
Refactor, consolidated  docker images belonging to lighthorizon processes to be under e`xp/lighthorizon/build`
documented [k8s MVP deployment](https://stellarorg.atlassian.net/wiki/spaces/HOR/pages/4393730346/k8s+deployment)

Closes #4493 

### Why

Enable deployment of lighthorizon web process onto infra/container orchestration layers.

### CODE REVIEW

File: exp/lighthorizon/main.go [hunk 1]
```diff
@@ -26,10 +27,17 @@ func main() {
...
 		"number of ledgers to store in the cache")
+	logLevelParam := flag.String("log-level", "info",
+		"logging level, info, debug, warn, error, panic, fatal, trace, default is info")
 	flag.Parse()
 
 	L := log.WithField("service", "horizon-lite")
-	L.SetLevel(log.InfoLevel)
+	logLevel, err := logrus.ParseLevel(*logLevelParam)
+	if err != nil {
+		logLevel = log.InfoLevel
```
Comment (Shaptic) [inline exp/lighthorizon/main.go:37]: nit: Maybe we could spit out a warning here? e.g. 
```suggestion
		logLevel = log.InfoLevel
		log.Warnf("Failed to parse -log-level '%s', defaulting to 'info'.", *logLevelParam)
```

File: exp/lighthorizon/build/k8s/lighthorizon_index.yml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    app: lighthorizon-pubnet-index
+  name: lighthorizon-pubnet-index-env
+data:
+  TXMETA_SOURCE: "s3://horizon-ledgermeta-prodnet-test"
+  INDEXES_SOURCE: "s3://horizon-index-prodnet-test"
+  NETWORK_PASSPHRASE: "Public Global Stellar Network ; September 2015"
+  START: "41809728"
+  END: "0"
+  WATCH: "true"
+  MODULES: "transactions,accounts"
```
Comment (Shaptic) [inline exp/lighthorizon/build/k8s/lighthorizon_index.yml:16]: We just need `accounts`, yeah?

File: exp/lighthorizon/build/index-batch/start [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+NETWORK_PASSPHRASE="${NETWORK_PASSPHRASE:=Public Global Stellar Network ; September 2015}"
+
+if [ "$RUN_MODE" == "reduce" ]; then
+    echo "Running Reduce, REDUCE JOBS: $REDUCE_JOB_COUNT MAP JOBS: $MAP_JOB_COUNT TARGET INDEX: $INDEX_TARGET"
+    /reduce 
+elif [ "$RUN_MODE" == "map" ]; then
+    echo "Running Map, TARGET INDEX: $INDEX_TARGET FIRST CHECKPOINT: $FIRST_CHECKPOINT"
+    /map 
+else
+    echo "error: undefined RUN_MODE env variable, must be map or reduce"
```
Comment (Shaptic) [inline exp/lighthorizon/build/index-batch/start:16]: recommend incl. the var in the message for easier debugging
```suggestion
    echo "error: undefined RUN_MODE env variable ('$RUN_MODE'), must be 'map' or 'reduce'"
```

File: exp/lighthorizon/build/build.sh [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    build_target
+    MODULE_NAME=web
+    build_target
+    MODULE_NAME=index-single
+    build_target
+    MODULE_NAME=ledgerexporter
+    build_target
+    ;;  
+*)
+    echo -n "unknown MODULE build parameter, must be one of all|index_batch|web|index_single|ledgerexporter"
```
Comment (Shaptic) [inline exp/lighthorizon/build/build.sh:48]: same here, incl. what was passed in
```suggestion
    echo -n "unknown MODULE build parameter ('$MODULE'), must be one of all|index_batch|web|index_single|ledgerexporter"
```

**Review by Shaptic — APPROVED** [review]
very nice refactor! seems like all the bases covered. maybe incl. a short README-esque file on what to run and how to run it depending on what you're trying to deploy and where?

File: exp/lighthorizon/build/k8s/lighthorizon_index.yml (hunk 2 above)
Comment (sreuland) [reply exp/lighthorizon/build/k8s/lighthorizon_index.yml:16]: I don't know either, I figured more is better, or should remove that now?

Comment (sreuland) [reply exp/lighthorizon/build/k8s/lighthorizon_index.yml]: ok, I removed it - https://github.com/stellar/go/pull/4519/commits/06030fe3c2f4c4e1f123e946e3c7a667aa372c17

**Comment by sreuland** [comment]
Hello @2opremio , before I merge, wanted to get your review on this, since it overlaps on indexer dockering, same as you are setting up on PR #4501 , ideally `./exp/lighthorizon/build/build.sh index-batch myrepo latest false` from this PR can provide the same result?

File: exp/lighthorizon/build/k8s/lighthorizon_index.yml (hunk 2 above)
Comment (Shaptic) [reply exp/lighthorizon/build/k8s/lighthorizon_index.yml]: We'll need lots of code changes before we use those indices anyway (different lookup style), so I think it's fine to drop it.

**Pull request merged**